### PR TITLE
feat(BE-39):Implement endpoint to check if a prompt exist in an image: upload

### PR DESCRIPTION
**Pull request by emmrys-jay**
### Endpoint
- Created endpoint to check if a text prompt exists in an image.
- Make a request to python microservice to return text description and a boolean.
- Upload image to s3 with its description.
- Store image as a hash of it's name on s3.
- Save imagepath and other metadata to mongodb.
- Return an object containing result of prompt check.

### Linear ticket
- https://linear.app/team-scale/issue/BE-39/implement-endpoint-to-allow-a-logged-in-user-check-if-a-prompt-exist

### CODE REVIEW

File: pkg/router/mine-service.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ func MineServiceUpload(r *gin.Engine, validate *validator.Validate, ApiVersion s
 	authUrl := r.Group(fmt.Sprintf("/api/%v", ApiVersion))
 	{
 		authUrl.POST("/mine-service/upload", mineservice.MineImageUpload)
+		authUrl.POST("/mine-service/upload/prompt", mineservice.PromptMineImageUpload)
```
Comment (kataszn) [inline pkg/router/mine-service.go]: upload-prompt is better than upload/prompt
upload is not a path, it is the action

File: internal/model/microservice.go [hunk 2]
```diff
@@ -3,3 +3,8 @@ package model
 type MicroserviceResponse struct {
 	Content string `json:"text_description"`
 }
+
+type MicroservicePromptResponse struct {
+	CheckResult bool   `json:"check_result"`
+	Content     string `json:"text_description"`
```
Comment (kataszn) [inline internal/model/microservice.go]: the microservice is supposed to return a "check_result" only, returning the content is another costly operation that is not really required

**Comment by emmrys-jay** [comment]
Alright, noted.

**Comment by emmrys-jay** [comment]
Also, what happens to saving the text content to mongodb if it doesn't return the text?

**Comment by emmrys-jay** [comment]
@usmansadik Issue has been resolved.

**Comment by emmrys-jay** [comment]
@usmansadik I made changes for the token functions you moved. You can merge now

**Pull request merged**